### PR TITLE
use ks setting

### PR DESCRIPTION
**Pull request by nguyenhoaidanh**
remove token list 
use ks setting

### CODE REVIEW

File: src/state/tokens/hooks.ts [hunk 1]
```diff
@@ -283,3 +287,58 @@ export function useTokenTransactions(address: string): Transaction[] | undefined
...
+  try {
+    const chainPromises = SUPPORTED_NETWORKS.map((chainId) => getWhitelistToken(chainId))
+    const data = await Promise.allSettled(chainPromises)
+    const TokenMap: WhiteListTokenMapByChain = {}
+    data.forEach((e, index) => {
+      TokenMap[SUPPORTED_NETWORKS[index]] = e.status === 'fulfilled' ? e.value : {}
+    })
+    return TokenMap
+  } catch (error) {
+    return {} as WhiteListTokenMapByChain
```
Comment (nguyenhuudungz) [inline src/state/tokens/hooks.ts:313]: Should not ignore the error here.

File: src/state/tokens/hooks.ts [hunk 2]
```diff
@@ -283,3 +287,58 @@ export function useTokenTransactions(address: string): Transaction[] | undefined
...
+
+function formatWhitelistToken(tokens: WrappedToken[]): WhiteListTokenMap {
+  return tokens.reduce<WhiteListTokenMap>((tokenMap: WhiteListTokenMap, token: WrappedToken) => {
+    const address: string = token.address || ''
+    if (address) tokenMap[address] = token
+    return tokenMap
+  }, {}) as WhiteListTokenMap
+}
+
+export const fetchAllTokenWhiteList = async (): Promise<WhiteListTokenMapByChain> => {
```
Comment (nguyenhuudungz) [inline src/state/tokens/hooks.ts:303]: Better func name: `fetchAllWhitelistToken`.

File: src/state/tokens/hooks.ts [hunk 3]
```diff
@@ -283,3 +287,58 @@ export function useTokenTransactions(address: string): Transaction[] | undefined
   // return data
   return transactions
 }
+
+export function useWhitelistTokenByChain(): WhiteListTokenMapByChain {
+  return useSelector<AppState>((state) => state.tokens.mapWhitelistToken) as WhiteListTokenMapByChain
+}
+
+function formatWhitelistToken(tokens: WrappedToken[]): WhiteListTokenMap {
```
Comment (nguyenhuudungz) [inline src/state/tokens/hooks.ts:295]: `formatWhitelistTokens`.

File: src/state/tokens/hooks.ts [hunk 4]
```diff
@@ -283,3 +287,58 @@ export function useTokenTransactions(address: string): Transaction[] | undefined
...
+      TokenMap[SUPPORTED_NETWORKS[index]] = e.status === 'fulfilled' ? e.value : {}
+    })
+    return TokenMap
+  } catch (error) {
+    return {} as WhiteListTokenMapByChain
+  }
+}
+
+// loop to fetch all whitelist token
+export async function getWhitelistToken(chainId: ChainId): Promise<WhiteListTokenMap> {
```
Comment (nguyenhuudungz) [inline src/state/tokens/hooks.ts:318]: `getWhitelistTokens`.

**Review by nguyenhuudungz — CHANGES_REQUESTED** [review]

File: src/state/tokens/hooks.ts (hunk 1 above)
Comment (nguyenhoaidanh) [reply src/state/tokens/hooks.ts:314]: okay, added console.error

File: src/state/tokens/hooks.ts [hunk 5]
```diff
@@ -283,3 +287,58 @@ export function useTokenTransactions(address: string): Transaction[] | undefined
...
+        const tokensResponse = data.tokens ?? []
+        tokens = tokens.concat(tokensResponse)
+        if (tokensResponse.length < pageSize || page >= maximumPage) break // out of tokens, and prevent infinity loop
+      }
+    } catch (error) {
+      return reject(`Failed to fetch list token of chainId ${chainId}`)
+    }
+    resolve(formatWhitelistToken(tokens))
+  })
+}
```
[the remark this answers is not in the document]
Comment (nguyenhoaidanh) [reply src/state/tokens/hooks.ts]: sound good, maybe I redundant here

File: src/components/CurrencyLogo/index.tsx [hunk 6]
```diff
@@ -39,10 +35,12 @@ const CurrencyLogo: React.FunctionComponent<CurrencyLogoType> = ({
...
 
-  const tokenlist = useCombinedActiveList()?.[activeNetwork.chainId]
+  const tokenlist = useWhitelistTokenByChain()?.[activeNetwork.chainId]
+
   const URI = useMemo(() => {
-    if (checkSummed && tokenlist?.[checkSummed]) {
-      return tokenlist?.[checkSummed].token.logoURI
+    const tokenInfo = checkSummed ? tokenlist?.[checkSummed] || tokenlist?.[checkSummed.toLowerCase()] : undefined
+    if (checkSummed && tokenInfo) {
+      return tokenInfo?.logoURI || undefined
```
[the remark this answers is not in the document]
Comment (nguyenhoaidanh) [reply src/components/CurrencyLogo/index.tsx]: updated all, continue to review bro

**Review by hungdoansy — APPROVED** [review]

**Review by nguyenhuudungz — APPROVED** [review]

**Review by namgold — APPROVED** [review]

**Pull request merged**